tslint.json

```diff
@@ -0,0 +1,11 @@
+{
+  "extends": "tslint-config-airbnb",
+  "rules": {
+    "variable-name": false,
+    "no-this-assignment": false,
+    "import-name": false,
+    "max-line-length": false,
+    "no-else-after-return": false,
+    "function-name": false
```

Comment:
Generally speaking I think it's good for us to limit the number of project-wide rule exceptions to best conform to common and popular styles.

`no-else-after-return` is one rule we'd agreed to discard earlier. After looking at these rules closely, I'm fine with keeping `variable-name": false` and `"import-name": false` because it allows us to keep our convention of upper case naming for classes and there seems to be a number of places where these apply.  For the others, I'd prefer to keep them out of the project-wide config, either disable them on a per-file or per-line basis or just edit the code to conform. `no-this-assignment` in particular is a useful rule to abide by, I think. We could also perhaps increase the limit for max-line-length, 150 characters fits easily on my default layout so personally I'm fine with `"max-line-length": [true, 150]`. Input from @moshababo would be good since we're gonna live with these rules for a while.